src/Pages/CustomDatePhoto.js

```diff
@@ -0,0 +1,107 @@
+    const [errorMessage, setErrorMessage] = useState("");
+
+    const url = `https://api.nasa.gov/planetary/apod?api_key=${process.env.REACT_APP_NASA_API_KEY}&date=`;
+
+    useEffect(() => {
+        let isMounted = true;
+        const getPhoto = async () => {
+        setLoading(true);
+        setError(false);
+        setVideo(false);
+        const newUrl = url + date.target.value;
+        try {
+            const response = await axios.get(newUrl);
+            if (isMounted){
+                if(response.data.media_type === "video"){
+                    setVideo(true);
+                    setData(response.data);
+                }else{
+                    setData(response.data);
+                }
+            }
+         } catch (error) {
+            setError(true);
+        }
+        setLoading(false);
+    };
+
+    getPhoto();
+
+    return () => { isMounted = false };
+
+    }, [date])// eslint-disable-line react-hooks/exhaustive-deps
+
+    const preSetDate = (e) => {
+        setDate(e);
+        let today = new Date().toISOString().substring(0, 10);
+        if(e.target.value > today) {
+            setError(true);
+            setErrorMessage('The selected date is in the future, please select today or a date in the past.');
+        }
+    }
+
+    if(!data) {
+        return(
+            <div>
+                <StyledInput type="date" id="date" value={date.target.value} onChange={preSetDate}></StyledInput>
+            </div>
+        )
+    }
+
+    return (
+        <>
+        {loading ? (
+            <div>
+                <p>Loading...</p>
+            </div>
+        ) : error ? (
+            <div>
+                 <StyledInput type="date" id="date" value={date.target.value} selected={date} onChange={preSetDate}></StyledInput>
+                <p>{errorMessage}</p>
+            </div>
+        ) : video ? (
+            <div>
+                <StyledInput type="date" id="date" data-testid="date-input" value={date.target.value} selected={date} onChange={preSetDate} />
+                <br/><br/>
+                <StyledImgContainer>
+                    <StyledIFrame data-testid="img" title={data.title} src={data.url}></StyledIFrame>
+                </StyledImgContainer>
+                <StyledTitle>{data?.title}</StyledTitle>
+                <StyledDescription>{data?.explanation}</StyledDescription>
+            </div>
+        ) : (
+            <div>
+                <StyledInput type="date" id="date" data-testid="date-input" value={date.target.value} selected={date} onChange={preSetDate} />
+                <br/><br/>
+                <StyledImgContainer>
+                    <StyledImg data-testid="img" src={data?.url}></StyledImg>
+                    <StyledImgText>{data?.copyright} ®</StyledImgText>
+                </StyledImgContainer>
+                <StyledTitle>{data?.title}</StyledTitle>
+                <StyledDescription>{data?.explanation}</StyledDescription>
+            </div>
+        )}
```

Comment:
You can delete some duplicity
```suggestion
     
            <div>
                <StyledInput type="date" id="date" data-testid="date-input" value={date.target.value} selected={date} onChange={preSetDate} />
                <br/><br/>
                <StyledImgContainer>
                video ?
                    <StyledIFrame data-testid="img" title={data.title} src={data.url}></StyledIFrame>
                    :
                    <StyledImg data-testid="img" src={data?.url}></StyledImg>
                    <StyledImgText>{data?.copyright} ®</StyledImgText>
                </StyledImgContainer>
                <StyledTitle>{data?.title}</StyledTitle>
                <StyledDescription>{data?.explanation}</StyledDescription>
            </div>
        ) 
```